world/petitions/models.py

```diff
@@ -10,7 +10,87 @@
 
 from server.utils.exceptions import PayError
 from .exceptions import PetitionError
+from web.character.models import (Clue, SearchTag)
+from world.dominion.models import PlayerOrNpc
+from world.dominion.general_dominion_commands import CmdPatronage
+
+class Interest(SharedMemoryModel):
+    name = models.CharField(max_length=30)
+    wantedads = models.ManyToManyField("WantedAd", related_name="wanted_ads")
+
+
+class WantedAd(SharedMemoryModel):
+    """This is an advertisement for wanting an RP partner"""
+    MARRIAGE = 0
+    SPONSOR = 1
+    PROTEGE = 2
+    PLOT = 3
+    TIME = 4
+    MATCH_TYPES = ((MARRIAGE, "Marriage"),(SPONSOR,"Sponsor"),(PROTEGE,"Protege"),(PLOT,"Plot"),(TIME,"Time"))
+    playtime_start = models.PositiveIntegerField(default=0)
+    playtime_end = models.PositiveIntegerField(default=0)
+    prefer_weekend = models.BooleanField(default=False)
+    interests = models.ManyToManyField(Interest)
+    clues = models.ManyToManyField(Clue)
```

Comment:
You'll wanna set `blank=True` for the M2M fields if they're not required, so there's not validation errors in django admin if we tried to edit something and left them blank. I normally use the string literal way of specifying models for fields (like `"character.Clue"`) because of trying to avoid having models files import one another, which can lead to circular import errors or random errors when trying to load models during django's startup process.